### DISCUSSION TITLE
Use custom email password UI

### DISCUSSION THREAD

**Discussion by KetulCodiste** [Web3Auth/Web3Auth#512] [Help]
I am auth0 for the user authenticaiton. To create the web3auth instance, I have used Web3AuthCore and OpenloginAdapter. For login:

`
await web3auth.connectTo(
      WALLET_ADAPTERS.OPENLOGIN,
      { 
        loginProvider: "jwt",
        extraLoginOptions: {
          domain: "My_domain",
          verifierIdField: "sub", 
        }
      }
    )
`

On calling the above code, A popup is getting open. The other option is to redirect which I can configure while creating the web3auth instance using uxMode in Openloginadapter. But there is no other option other than this two. What i want to achieve is, I do not want a pop up. Instead, I want to create my own UI, where I can add the username and password, and after clicking the login button, the user gets authenticated and redirect to my application. Means I do not want redirect or popup. Reminder: i am using auth0.

**Comment by rinzler99** [comment]
As you mentioned user gets authenticated and "redirects" to your application after entering username and password means you can use redirect mode.
In order to use your own UI you can configure auth0 tenant with custom ui.

https://auth0.com/docs/customize/universal-login-pages/universal-login-page-templates

**Reply by KetulCodiste** [reply]
I do not want to use redirect mode
I have a website and a login page with email and password fields.
with web3auth + auth0, I would have to remove the email and password fields from my UI signin page and just show a button instead of email and password fields, then after clicking that button, web3auth will either show me popup of auth0 or redirects me to auth0.
But I do not want this behaviour. currently I have email and password fields in my signin page. I want that user can enter there email and password in my signin page only, clicks the signin button and gets logged in if the credentials are correct.

**Reply by rinzler99** [reply]
Right now that is not possible, you need to use either redirect flow or pop up flow. That's mandatory field in the openlogin adapter configuration.

**Reply by yashovardhan** [reply]
Adding to what @rinzler99 mentioned, this redirection to openlogin.com via a pop up or redirect is mandatory for construction for the users keys. To make Web3Auth non custodial, we construct the keys in the frontend. This is happening on openlogin.com which hosts our offchain multi sig implementation. 

If you want to totally skip openlogin.com, you can use [Web3Auth Self Host](https://web3auth.io/docs/sdk/self-host/installation) instead.
